src/Swift.Bindings/tests/IntegrationTests/SmokeTests/SmokeTests.cs

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Xunit;
+using Swift.SmokeTests;
+
+namespace BindingsGeneration.Tests
+{
+    public class SmokeTests : IClassFixture<SmokeTests.TestFixture>
```

Comment:
I would see it the other way around. Lets consider some example lets say simple frozen structs. I think the test setup should look like this:
1. We have a set of "integration" unit tests - basically small pieces of swift code which we project into c# and verify that they work. So for frozen structs we can have e.g.. "CallsInstanceMethodOnFrozenStruct" or "ConstructsNewNonFrozenStruct" etc.
2. Then for bunch of features (lets say all structs code) we have some stress test where we try to generate all the possible interactions of those in order to verify we have not missed some scenarios. Those should not be bound to "FrozenStructs" category, but rather operate on broader scope to test more interactions
3. Framework tests - where we apply the projections to real examples

Reply:
Ok, I wasnt precise. I agree that things living in "UnitTests" should not depend of Swift Code. But this means that UnitTests will most likely not test projections in any way, except for some Swift.Runtime types)

So talking strictly about stuff which lives in `IntegrationTests` space. I see it like this
```
integration tests
├── functional tests
│   ├── StructTests
│   ├── FuncTests
│   └── etc
├── stress tests
│   └── StructsStressTests \\ or even broader
└── framework tests
``` 

where functional tests will contain (mostly) hand-written tests which should cover (ideally all) use cases. Whereas stress test can be source generated and serve as a safety net to catch hard to spot bugs.

Reply:
> Ok, I wasnt precise. I agree that things living in "UnitTests" should not depend of Swift Code. But this means that UnitTests will most likely not test projections in any way, except for some Swift.Runtime types)

Correct, it will not test projections but the projection tooling components.

> So talking strictly about stuff which lives in IntegrationTests space. I see it like this

Sounds good -- Let's reflect it in the PR.